File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
class Solution {
    public int check(int n){
        switch(n){
            case 6:
                return 1;
                
            case 5:
                return 2;
                
            case 4:
                return 3;
                
            case 3:
                return 4;
                
            case 2:
                return 5;
               
            default:
                return 6;
                
                
        }
    }
    public int[] solution(int[] lottos, int[] win_nums) {
        
        int correctCount=0;
        int zeroCount=0;
        for(int i : lottos){
            for(int j:win_nums){
                if(i==j){
                    correctCount++;
                }
            }
            if(i==0){
                zeroCount++;
            }
        }
        int min=correctCount;
        int max=correctCount+zeroCount;
        
        int[] answer = {check(max),check(min)};
        return answer;
    }
}

File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):

    rank=[6,6,5,4,3,2,1]

    count_zero = lottos.count(0)
    answer = 0
    for i in win_nums:
        if i in lottos:
            answer += 1
    return rank[count_zero + answer],rank[answer]